uber/models.py

```diff
@@ -1467,32 +1471,40 @@ def badge_cost_without_promo_code(self):
         return self.calculate_badge_cost(use_promo_code=False)
 
     def calculate_badge_cost(self, use_promo_code=True):
-        registered = self.registered_local if self.registered else None
         if self.paid == c.NEED_NOT_PAY:
             return 0
         elif self.overridden_price is not None:
             return self.overridden_price
-        elif self.is_dealer:
-            cost = c.DEALER_BADGE_PRICE
-        elif self.badge_type == c.ONE_DAY_BADGE:
-            cost = c.get_oneday_price(registered)
-        elif self.is_presold_oneday:
-            cost = c.get_presold_oneday_price(self.badge_type)
-        elif self.badge_type in c.BADGE_TYPE_PRICES:
-            cost = int(c.BADGE_TYPE_PRICES[self.badge_type])
-        elif self.age_discount != 0:
-            cost = max(0, c.get_attendee_price(registered) + self.age_discount)
-        elif self.group and self.paid == c.PAID_BY_GROUP:
-            cost = c.get_attendee_price(registered) - c.GROUP_DISCOUNT
+        elif self.base_badge_price:
+            cost = self.base_badge_price
         else:
-            cost = c.get_attendee_price(registered)
+            cost = self.new_badge_cost
 
         if c.BADGE_PROMO_CODES_ENABLED and self.promo_code and use_promo_code:
             return self.promo_code.calculate_discounted_price(cost)
         else:
             return cost
 
     @property
+    def new_badge_cost(self):
+        # What this badge would cost if it were new, i.e., not taking into account special overrides
+        registered = self.registered_local if self.registered else None
+        if self.is_dealer:
+            return c.DEALER_BADGE_PRICE
+        elif self.badge_type == c.ONE_DAY_BADGE:
+            return c.get_oneday_price(registered)
+        elif self.is_presold_oneday:
+            return c.get_presold_oneday_price(self.badge_type)
+        elif self.badge_type in c.BADGE_TYPE_PRICES:
+            return int(c.BADGE_TYPE_PRICES[self.badge_type])
+        elif self.age_discount != 0:
+            return max(0, c.get_attendee_price(registered) + self.age_discount)
```

Comment:
Should the age discount logic be checked first? It seems like it won't get checked if any of the things above are true.